polyA/substitution_matrix.py

```diff
@@ -99,8 +123,20 @@ def load_substitution_matrices(
         except StopIteration:
             break
 
+        # FIXME: will this break if there is no next line?
+        next_line = next(file)
```

Comment:
Yep, use the trick above where you catch `StopIteration`, which is the exception that will be raised when the iterator is exhausted.